src/main/java/edu/wpi/grip/ui/pipeline/input/InputSocketViewFactory.java

```diff
@@ -0,0 +1,64 @@
+package edu.wpi.grip.ui.pipeline.input;
+
+import com.google.common.eventbus.EventBus;
+import edu.wpi.grip.core.InputSocket;
+import edu.wpi.grip.core.SocketHint;
+
+import java.util.List;
+
+/**
+ * Factory for constructing editable controls for input sockets
+ */
+public class InputSocketViewFactory {
+
+    /**
+     * Create an instance of {@link InputSocketView} appropriate for the given socket.
+     */
```

Comment:
Doesn't really matter, does it? A lot of methods we have don't have a tag for every parameter because it's better to write comments that are helpful to people than ones that blindly follow rules.